homeassistant/components/shelly/sensor.py

```diff
@@ -907,6 +907,9 @@ class RestSensorDescription(RestEntityDescription, SensorEntityDescription):
         device_class=SensorDeviceClass.SIGNAL_STRENGTH,
         state_class=SensorStateClass.MEASUREMENT,
         entity_registry_enabled_default=False,
+        removal_condition=lambda config, _status, key: (
+            config[key]["sta"]["enable"] is False
+        ),
```

Comment:
I think it would be worth adding a test to confirm the removing of the entity. Example here https://github.com/home-assistant/core/blob/f768dd876181160e2e0bf6a0ad54d4df9442e660/tests/components/shelly/test_binary_sensor.py#L184-L198

Reply:
I though that we are already testing the removal condition (even if with a different `binary_sensor`)

Reply:
Yeah, you’re right, we already have tests for removal condition.